ghost/external-media-inliner/lib/ExternalMediaInliner.js

```diff
@@ -63,21 +73,32 @@ class ExternalMediaInliner {
      * @param {Object} response - response from request
      * @returns {Object}
      */
-    #extractFileDataFromResponse(requestURL, response) {
-        const headers = response.headers;
-        const contentType = headers['content-type'];
+    async extractFileDataFromResponse(requestURL, response) {
+        let extension;
+
+        try {
+            const fileInfo = await FileType.fromBuffer(response.body);
+            extension = fileInfo.ext;
```

Comment:
This bit might return undefined according to docs:
> Or undefined when there is no match.

Worth putting the content-type fallback should account for that case too

Reply:
If it returns undefined, the try/catch fails with `Cannot read properties of undefined (reading 'ext')`, so I think this already falls back to using the content-type/extension correctly.

Reply:
That's against of a strict "do not control the logic flow with exceptions" but well, ok 😄 maybe just leave a comment about why/when the exception is thrown haha because if someone else reads this it would be slightly weird

Reply:
Fair comment! I've added a comment explaining it.